src/qt/diagnosticsdialog.h

```diff
@@ -0,0 +1,72 @@
+#ifndef DIAGNOSTICSDIALOG_H
+#define DIAGNOSTICSDIALOG_H
+
+#include <QDialog>
+#include <QtNetwork>
+#include <QtConcurrent>
+
+#include "main.h"
+#include "util.h"
+#include <boost/algorithm/string.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/date_time/posix_time/posix_time.hpp>
+
+extern std::string ExtractXML(std::string XMLdata, std::string key, std::string key_end);
```

Comment:
No, extern go away.

Reply:
agreed no need for these externs.

Things to know: 

Functions that can be removed as they in header files:
main.h:
ExtractXML
const CBlockIndex* GetLastBlockIndex
std::map<std::string, int64_t> mvApplicationCacheTimestamp

util.h: 
GetAdjustedTime
ToString 
RoundToString

Remove extern for:
GetListOf
TimestampToHRDate
PreviousBlockAge
GetDifficulty

* add include for boinc.h for function GetBoincDataDir and remove declaration (probably better this way as its not external anywhere and anywhere used boinc.h is included) 

also personal preference for myself is I don't include headers and functions that are being called externally in a header file as its rarely needed.

Reply:
Yes. As little in header file as possible. All includes not needed in the header, shoudl be moved to cpp.